### PR TITLE
Create To Do List application

### PR DESCRIPTION
**Pull request by NitBravoA92**
# 💯 In this Milestone, I:
✅ Installed and set up Webpack, Webpack dev server, HTML Webpack plugin, and a CSS Loader
✅ Created an index.html file in the src folder, with all the main markup of the list
✅ Created a style.css file in the src folder with all the CSS styles for the To-Do List
✅ Created an index.js file in the src folder. It is the entry point of the application
✅ Created a modules folder  in the src folder to store all the modules of the application
✅ Created the module listTask.js with all the functions to list all tasks on the page
✅ Created the module renderTask.js with all the functions to iterate over the tasks array and populate an HTML list item element for each task
✅ Created the module createTask.js with all the functions to create and insert a new task item in the list
✅ Created the module updateTask.js with all the functions to update tasks from the list
✅ Created the module deleteTask.js with all the functions to delete tasks from the list
✅ Created the module Utils.js with all the functions to work with local storage, generate task markup, generate objects, clone, and check array length. They are supporter functions to reuse code.
✅ I generated the final application files in the ./dist folder using Webpack

# ⭐To the code reviewer 👨‍💻
💝 I really thank 🙇‍♂️ you, dear code reviewer 👨‍🎨 for dedicating your precious 🥇 time ⌚ to check this PR 🧾 .

I kindly ask 🙏 dear code reviewer 🤵 that if there is any kind of issue 🦯 in this project, please do list 📃 them in a descriptive 💡 manner and give your best suggestions 🎁 if needed.

If you think there are some big issues that are essential to be changed ♻️ please kindly contact 📞 me through Zoom or Slack, I am available on Microverse UTC-6 time zone.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>To Do List</title>
+</head>
+<body>
+  <main>
+    <section id="app">
+      <h2 id="title"><span>Today's To Do</span>
```
Comment (AndreaM2429) [inline src/index.html]: 📍 [Optional] Kindly use a `h1` tag inted of a `h2` tag. According to the best HTML practices a page must has a unique h1 relate to the main content.

File: src/modules/Utils.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  <div class="task-options">
+    <div class="drag-icon">
+      <span class="dot"></span>
+      <span class="dot"></span>
+      <span class="dot"></span>
+    </div>
+  </div>
+  <button type="button" class="remove-task" disabled>
+    <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="#bebebe" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><polyline points="3 6 5 6 21 6"></polyline><path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path><line x1="10" y1="11" x2="10" y2="17"></line><line x1="14" y1="11" x2="14" y2="17"></line></svg>
+  </button>
```
Comment (AndreaM2429) [inline src/modules/Utils.js]: 📍 [Optional] As you are using Webpack, you could charge the the images using it changing the configuration on your `webpack.config.js`. Also you can use alternatives to use images with [unicodes](https://www.toptal.com/designers/htmlarrows/)

File: src/modules/createTask.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  const btnAddTask = document.querySelector('#btnAddTask');
+  inputDescription.addEventListener('keydown', (event) => {
+    if (event.code === 'Enter') {
+      createTask(inputDescription);
+    }
+  });
+  btnAddTask.addEventListener('click', () => {
+    createTask(inputDescription);
+  });
+};
```
Comment (AndreaM2429) [inline src/modules/createTask.js]: 📍 [Optional] Please keep in mind that if you use a `keydown` event the action will ocurr every time the user press a key so I suggest you use a `timer` to delete the number of times that the evet will be triggered or also you can change the event to `focusout`

**Review by AndreaM2429 — APPROVED** [review]
### Excellent job @NitBravoA92!
You make an excellent job bulding this, so you're Approve! 🎆🎉🎊

Congratulations your project is completet you can merge it 🐱‍👤.



### Highlights
🎯DRY
🎯Great file distibution
🎯Excellent use of JavaScript practices
🎯Descriptive pull request

### Optional suggestions

Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!👏👏👏


> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: src/index.html (hunk 1 above)
Comment (NitBravoA92) [reply src/index.html]: Hi @AndreaM2429 I really appreciate the time you have spent reviewing my code and leaving me your suggestions. Thank you so much. I'm going to take note of this change and apply it to make my code much better.

File: src/modules/Utils.js (hunk 2 above)
Comment (NitBravoA92) [reply src/modules/Utils.js]: @AndreaM2429 Thank you very much for this valuable suggestion. This is a good approach, which I am definitely going to apply.

File: src/modules/createTask.js (hunk 3 above)
Comment (NitBravoA92) [reply src/modules/createTask.js]: @AndreaM2429 It is an excellent suggestion. Thank you so much. I'm going to try using a `timer` to improve the performance of this functionality.

**Pull request merged**